### ISSUE TITLE
jobservice crash

### ISSUE THREAD

**Issue by mikeshi80** [vmware/harbor#2698]
I am using harbor 1.1.1, and since July 1, the jobservice failed to replicate the images.

I checked the jobservice.log file, and got the following errors:
```
Jul  4 16:57:22 172.18.0.1 jobservice[10132]: 2017-07-04T08:57:22Z [DEBUG] [statemachine.go:88]: Job id: 18474, transition succeeded, current state: running
Jul  4 16:57:22 172.18.0.1 jobservice[10132]: 2017-07-04T08:57:22Z [DEBUG] [statemachine.go:97]: Job id: 18474, next state from handler: _continue
Jul  4 16:57:22 172.18.0.1 jobservice[10132]: 2017-07-04T08:57:22Z [DEBUG] [statemachine.go:109]: Job id: 18474, Continue to state: initialize
Jul  4 16:57:22 172.18.0.1 jobservice[10132]: 2017-07-04T08:57:22Z [DEBUG] [statemachine.go:61]: Job id: 18474, transiting from State: running, to State: initialize
Jul  4 16:57:22 172.18.0.1 jobservice[10132]: 2017-07-04T08:57:22Z [DEBUG] [statemachine.go:88]: Job id: 18455, transition succeeded, current state: running
Jul  4 16:57:22 172.18.0.1 jobservice[10132]: 2017-07-04T08:57:22Z [DEBUG] [statemachine.go:97]: Job id: 18455, next state from handler: _continue
Jul  4 16:57:22 172.18.0.1 jobservice[10132]: 2017-07-04T08:57:22Z [DEBUG] [statemachine.go:109]: Job id: 18455, Continue to state: initialize
Jul  4 16:57:22 172.18.0.1 jobservice[10132]: 2017-07-04T08:57:22Z [DEBUG] [statemachine.go:61]: Job id: 18455, transiting from State: running, to State: initialize
Jul  4 16:57:22 172.18.0.1 jobservice[10132]: 2017-07-04T08:57:22Z [DEBUG] [statemachine.go:88]: Job id: 18456, transition succeeded, current state: running
Jul  4 16:57:22 172.18.0.1 jobservice[10132]: 2017-07-04T08:57:22Z [DEBUG] [statemachine.go:97]: Job id: 18456, next state from handler: _continue
Jul  4 16:57:22 172.18.0.1 jobservice[10132]: 2017-07-04T08:57:22Z [DEBUG] [statemachine.go:109]: Job id: 18456, Continue to state: initialize
Jul  4 16:57:22 172.18.0.1 jobservice[10132]: 2017-07-04T08:57:22Z [DEBUG] [statemachine.go:61]: Job id: 18456, transiting from State: running, to State: initialize
Jul  4 16:57:23 172.18.0.1 jobservice[10132]: 2017-07-04T08:57:23Z [DEBUG] [statemachine.go:88]: Job id: 18456, transition succeeded, current state: initialize
Jul  4 16:57:23 172.18.0.1 jobservice[10132]: 2017-07-04T08:57:23Z [DEBUG] [statemachine.go:118]: Job id: 18456, next state from handler: check
Jul  4 16:57:23 172.18.0.1 jobservice[10132]: 2017-07-04T08:57:23Z [DEBUG] [statemachine.go:61]: Job id: 18456, transiting from State: initialize, to State: check
Jul  4 16:57:23 172.18.0.1 jobservice[10132]: 2017-07-04T08:57:23Z [DEBUG] [statemachine.go:88]: Job id: 18455, transition succeeded, current state: initialize
Jul  4 16:57:23 172.18.0.1 jobservice[10132]: 2017-07-04T08:57:23Z [DEBUG] [statemachine.go:118]: Job id: 18455, next state from handler: check
Jul  4 16:57:23 172.18.0.1 jobservice[10132]: 2017-07-04T08:57:23Z [DEBUG] [statemachine.go:61]: Job id: 18455, transiting from State: initialize, to State: check
Jul  4 16:57:23 172.18.0.1 jobservice[10132]: panic: runtime error: invalid memory address or nil pointer dereference
Jul  4 16:57:23 172.18.0.1 jobservice[10132]: [signal SIGSEGV: segmentation violation code=0x1 addr=0x60 pc=0x72955d]
Jul  4 16:57:23 172.18.0.1 jobservice[10132]:
Jul  4 16:57:23 172.18.0.1 jobservice[10132]: goroutine 16 [running]:
Jul  4 16:57:23 172.18.0.1 jobservice[10132]: panic(0x9c6d60, 0xc42000c060)
Jul  4 16:57:23 172.18.0.1 jobservice[10132]: #011/usr/local/go/src/runtime/panic.go:500 +0x1a1
Jul  4 16:57:23 172.18.0.1 jobservice[10132]: github.com/vmware/harbor/src/jobservice/replication.(*Checker).enter(0xc420370060, 0xf2bca0, 0x414f50, 0xc420043a50, 0x30272089)
Jul  4 16:57:23 172.18.0.1 jobservice[10132]: #011/go/src/github.com/vmware/harbor/src/jobservice/replication/transfer.go:195 +0x19d
Jul  4 16:57:23 172.18.0.1 jobservice[10132]: github.com/vmware/harbor/src/jobservice/replication.(*Checker).Enter(0xc420370060, 0xc420362840, 0xa57f84, 0x5, 0xc420464658)
Jul  4 16:57:23 172.18.0.1 jobservice[10132]: #011/go/src/github.com/vmware/harbor/src/jobservice/replication/transfer.go:179 +0x2f
Jul  4 16:57:23 172.18.0.1 jobservice[10132]: github.com/vmware/harbor/src/jobservice/job.(*SM).EnterState(0xc420311c00, 0xa57f84, 0x5, 0x2, 0x2, 0x0, 0x0)
Jul  4 16:57:23 172.18.0.1 jobservice[10132]: #011/go/src/github.com/vmware/harbor/src/jobservice/job/statemachine.go:80 +0x321
Jul  4 16:57:23 172.18.0.1 jobservice[10132]: github.com/vmware/harbor/src/jobservice/job.(*SM).Start(0xc420311c00, 0xa5aa5d, 0x7)
Jul  4 16:57:23 172.18.0.1 jobservice[10132]: #011/go/src/github.com/vmware/harbor/src/jobservice/job/statemachine.go:117 +0x23a
Jul  4 16:57:23 172.18.0.1 jobservice[10132]: github.com/vmware/harbor/src/jobservice/job.(*Worker).handleRepJob(0xc4203337c0, 0x4818)
Jul  4 16:57:23 172.18.0.1 jobservice[10132]: #011/go/src/github.com/vmware/harbor/src/jobservice/job/workerpool.go:96 +0x206
Jul  4 16:57:23 172.18.0.1 jobservice[10132]: github.com/vmware/harbor/src/jobservice/job.(*Worker).Start.func1(0xc4203337c0)
Jul  4 16:57:23 172.18.0.1 jobservice[10132]: #011/go/src/github.com/vmware/harbor/src/jobservice/job/workerpool.go:63 +0x204
Jul  4 16:57:23 172.18.0.1 jobservice[10132]: created by github.com/vmware/harbor/src/jobservice/job.(*Worker).Start
Jul  4 16:57:23 172.18.0.1 jobservice[10132]: #011/go/src/github.com/vmware/harbor/src/jobservice/job/workerpool.go:71 +0x3f
Jul  4 16:57:23 172.18.0.1 jobservice[10132]: panic: runtime error: invalid memory address or nil pointer dereference
Jul  4 16:57:23 172.18.0.1 jobservice[10132]: [signal SIGSEGV: segmentation violation code=0x1 addr=0x60 pc=0x72955d]
Jul  4 16:57:23 172.18.0.1 jobservice[10132]:
Jul  4 16:57:23 172.18.0.1 jobservice[10132]: goroutine 50 [running]:
Jul  4 16:57:23 172.18.0.1 jobservice[10132]: panic(0x9c6d60, 0xc42000c060)
Jul  4 16:57:23 172.18.0.1 jobservice[10132]: #011/usr/local/go/src/runtime/panic.go:500 +0x1a1
Jul  4 16:57:23 172.18.0.1 jobservice[10132]: github.com/vmware/harbor/src/jobservice/replication.(*Checker).enter(0xc420342080, 0xf2bca0, 0x414f50, 0xc4203c7a50, 0x2603e8e7)
Jul  4 16:57:23 172.18.0.1 jobservice[10132]: #011/go/src/github.com/vmware/harbor/src/jobservice/replication/transfer.go:195 +0x19d
Jul  4 16:57:23 172.18.0.1 jobservice[10132]: github.com/vmware/harbor/src/jobservice/replication.(*Checker).Enter(0xc420342080, 0xc42033cc60, 0xa57f84, 0x5, 0xc42045caa8)
Jul  4 16:57:23 172.18.0.1 jobservice[10132]: #011/go/src/github.com/vmware/harbor/src/jobservice/replication/transfer.go:179 +0x2f
Jul  4 16:57:23 172.18.0.1 jobservice[10132]: github.com/vmware/harbor/src/jobservice/job.(*SM).EnterState(0xc420311c70, 0xa57f84, 0x5, 0x2, 0x2, 0x0, 0x0)
Jul  4 16:57:23 172.18.0.1 jobservice[10132]: #011/go/src/github.com/vmware/harbor/src/jobservice/job/statemachine.go:80 +0x321
Jul  4 16:57:23 172.18.0.1 jobservice[10132]: github.com/vmware/harbor/src/jobservice/job.(*SM).Start(0xc420311c70, 0xa5aa5d, 0x7)
Jul  4 16:57:23 172.18.0.1 jobservice[10132]: #011/go/src/github.com/vmware/harbor/src/jobservice/job/statemachine.go:117 +0x23a
Jul  4 16:57:23 172.18.0.1 jobservice[10132]: github.com/vmware/harbor/src/jobservice/job.(*Worker).handleRepJob(0xc420333840, 0x4817)
Jul  4 16:57:23 172.18.0.1 jobservice[10132]: #011/go/src/github.com/vmware/harbor/src/jobservice/job/workerpool.go:96 +0x206
Jul  4 16:57:23 172.18.0.1 jobservice[10132]: github.com/vmware/harbor/src/jobservice/job.(*Worker).Start.func1(0xc420333840)
Jul  4 16:57:23 172.18.0.1 jobservice[10132]: #011/go/src/github.com/vmware/harbor/src/jobservice/job/workerpool.go:63 +0x204
Jul  4 16:57:23 172.18.0.1 jobservice[10132]: created by github.com/vmware/harbor/src/jobservice/job.(*Worker).Start
Jul  4 16:57:23 172.18.0.1 jobservice[10132]: #011/go/src/github.com/vmware/harbor/src/jobservice/job/workerpool.go:71 +0x3f
```

**Comment by mikeshi80** [comment]
I found the way the resolve this issue, I use `docker exec -it harbor-jobservice bash`, and use mysql to delete the job id 18455 row, restart jobservice container, and it came back to life.

Sorry, I forgot to leave the data of job id 18455 for further inspection.

**Comment by reasonerjt** [comment]
This is probably a dup? @ywk253100

**Comment by mikeshi80** [comment]
I found only new pushed images will be replicated, the elder will just be tagged as finished, in fact, it is not.

And now I update all jobs' status as 'retrying', and restart the jobservice container. Currently it works well, if there is any other issues, I will post here.

**Comment by mikeshi80** [comment]
Aha, if anyone push the image to harbor, the replication will be interrupted. I guess it is because the new generated job id is greater than old ones. Some rule was triggered.

**Comment by ywk253100** [comment]
Seems the cause of crach is the job can not find the project which the image belongs to and we have already a fix in the master branch to recover the errors.

**Comment by mikeshi80** [comment]
I upgraded jobservice to 1.1.2, it still exists. Is it included in 1.1.3?

**Comment by mikeshi80** [comment]
And I found another issue, I updated all status to "retrying", but I found after I restart jobservice, it will automatically stop when it processed about 1040 jobs. Is it some rule?

**Comment by reasonerjt** [comment]
@mikekim no it's not, it's in master only.
We'll try to reproduce and investigate the root cause.

**Comment by ywk253100** [comment]
It should be fixed.

**Comment by timgriffiths** [comment]
@ywk253100 i am having a very similar issue which release is this fixed in?

**Comment by wuzhihui1123** [comment]
my jobservice cann't start, the part log in below

=====
Jul 17 11:36:45 192.168.100.1 jobservice[50862]: 2017/07/17 03:36:45 #033[1;34m[asm_amd64.s:2086][I] http server Running on :80#033[0m
Jul 17 11:36:45 192.168.100.1 jobservice[50862]: 2017-07-17T03:36:45Z [DEBUG] [statemachine.go:61]: Job id: 522, transiting from State: pending, to State: running
Jul 17 11:36:45 192.168.100.1 jobservice[50862]: 2017-07-17T03:36:45Z [DEBUG] [statemachine.go:74]: Job id: 522, no handler found for state:pending, skip
Jul 17 11:36:45 192.168.100.1 jobservice[50862]: 2017-07-17T03:36:45Z [DEBUG] [statemachine.go:88]: Job id: 522, transition succeeded, current state: running
Jul 17 11:36:45 192.168.100.1 jobservice[50862]: 2017-07-17T03:36:45Z [DEBUG] [statemachine.go:97]: Job id: 522, next state from handler: _continue
Jul 17 11:36:45 192.168.100.1 jobservice[50862]: 2017-07-17T03:36:45Z [DEBUG] [statemachine.go:109]: Job id: 522, Continue to state: initialize
Jul 17 11:36:45 192.168.100.1 jobservice[50862]: 2017-07-17T03:36:45Z [DEBUG] [statemachine.go:61]: Job id: 522, transiting from State: running, to State: initialize
Jul 17 11:36:45 192.168.100.1 jobservice[50862]: 2017-07-17T03:36:45Z [DEBUG] [statemachine.go:88]: Job id: 522, transition succeeded, current state: initialize
Jul 17 11:36:45 192.168.100.1 jobservice[50862]: 2017-07-17T03:36:45Z [DEBUG] [statemachine.go:118]: Job id: 522, next state from handler: check
Jul 17 11:36:45 192.168.100.1 jobservice[50862]: 2017-07-17T03:36:45Z [DEBUG] [statemachine.go:61]: Job id: 522, transiting from State: initialize, to State: check
**Jul 17 11:36:45 192.168.100.1 jobservice[50862]: panic: runtime error: invalid memory address or nil pointer dereference**
Jul 17 11:36:45 192.168.100.1 jobservice[50862]: [signal SIGSEGV: segmentation violation code=0x1 addr=0x60 pc=0x72955d]
Jul 17 11:36:45 192.168.100.1 jobservice[50862]:
Jul 17 11:36:45 192.168.100.1 jobservice[50862]: goroutine 50 [running]:
Jul 17 11:36:45 192.168.100.1 jobservice[50862]: panic(0x9c6d60, 0xc42000c060)
Jul 17 11:36:45 192.168.100.1 jobservice[50862]: #011/usr/local/go/src/runtime/panic.go:500 +0x1a1
Jul 17 11:36:45 192.168.100.1 jobservice[50862]: github.com/vmware/harbor/src/jobservice/replication.(*Checker).enter(0xc42007a940, 0xf2bca0, 0x414f50, 0xc420049a50, 0x6ce4e08f)
Jul 17 11:36:45 192.168.100.1 jobservice[50862]: #011/go/src/github.com/vmware/harbor/src/jobservice/replication/transfer.go:195 +0x19d
Jul 17 11:36:45 192.168.100.1 jobservice[50862]: github.com/vmware/harbor/src/jobservice/replication.(*Checker).Enter(0xc42007a940, 0xc4204310e0, 0xa57f84, 0x5, 0xc420438308)
Jul 17 11:36:45 192.168.100.1 jobservice[50862]: #011/go/src/github.com/vmware/harbor/src/jobservice/replication/transfer.go:179 +0x2f
Jul 17 11:36:45 192.168.100.1 jobservice[50862]: github.com/vmware/harbor/src/jobservice/job.(*SM).EnterState(0xc42039a1c0, 0xa57f84, 0x5, 0x2, 0x2, 0x0, 0x0)
Jul 17 11:36:45 192.168.100.1 jobservice[50862]: #011/go/src/github.com/vmware/harbor/src/jobservice/job/statemachine.go:80 +0x321
Jul 17 11:36:45 192.168.100.1 jobservice[50862]: github.com/vmware/harbor/src/jobservice/job.(*SM).Start(0xc42039a1c0, 0xa5aa5d, 0x7)
Jul 17 11:36:45 192.168.100.1 jobservice[50862]: #011/go/src/github.com/vmware/harbor/src/jobservice/job/statemachine.go:117 +0x23a
Jul 17 11:36:45 192.168.100.1 jobservice[50862]: github.com/vmware/harbor/src/jobservice/job.(*Worker).handleRepJob(0xc42039c520, 0x20a)
Jul 17 11:36:45 192.168.100.1 jobservice[50862]: #011/go/src/github.com/vmware/harbor/src/jobservice/job/workerpool.go:96 +0x206
Jul 17 11:36:45 192.168.100.1 jobservice[50862]: github.com/vmware/harbor/src/jobservice/job.(*Worker).Start.func1(0xc42039c520)
Jul 17 11:36:45 192.168.100.1 jobservice[50862]: #011/go/src/github.com/vmware/harbor/src/jobservice/job/workerpool.go:63 +0x204
Jul 17 11:36:45 192.168.100.1 jobservice[50862]: created by github.com/vmware/harbor/src/jobservice/job.(*Worker).Start
Jul 17 11:36:45 192.168.100.1 jobservice[50862]: #011/go/src/github.com/vmware/harbor/src/jobservice/job/workerpool.go:71 +0x3f

**Comment by zhoulouzi** [comment]
i got the same err. the jobservice failed to replicate the images 
i have a image which path is "library/osixia/phpldapadmin"
delete the row is useful for me , but could someone tell me whats wrong.

some info:
row:
| 479 | canceled |         6 | library/osixia/phpldapadmin          | transfer  |                    | 2017-07-20 08:32:38 | 2017-07-20 08:55:43 |
log:

Jul 20 16:55:36 172.18.0.1 jobservice[21101]: 2017-07-20T08:55:36Z [DEBUG] [transport.go:50]: 200 | GET http://registry:5000/v2/library/osixia/phpldapadmin/tags/list
Jul 20 16:55:36 172.18.0.1 jobservice[21101]: 2017-07-20T08:55:36Z [DEBUG] [statemachine.go:88]: Job id: 479, transition succeeded, current state: initialize
Jul 20 16:55:36 172.18.0.1 jobservice[21101]: 2017-07-20T08:55:36Z [DEBUG] [statemachine.go:118]: Job id: 479, next state from handler: check
Jul 20 16:55:36 172.18.0.1 jobservice[21101]: 2017-07-20T08:55:36Z [DEBUG] [statemachine.go:61]: Job id: 479, transiting from State: initialize, to State: check
Jul 20 16:55:36 172.18.0.1 jobservice[21101]: 2017-07-20T08:55:36Z [DEBUG] [transport.go:50]: 202 | POST https://reg-bj.xiaoneng.cn/v2/library/rancher-server/blobs/uploads/
Jul 20 16:55:36 172.18.0.1 jobservice[21101]: panic: runtime error: invalid memory address or nil pointer dereference
Jul 20 16:55:36 172.18.0.1 jobservice[21101]: [signal SIGSEGV: segmentation violation code=0x1 addr=0x60 pc=0x72955d]
Jul 20 16:55:36 172.18.0.1 jobservice[21101]:
Jul 20 16:55:36 172.18.0.1 jobservice[21101]: goroutine 24 [running]:
Jul 20 16:55:36 172.18.0.1 jobservice[21101]: panic(0x9c6d60, 0xc42000c060)
Jul 20 16:55:36 172.18.0.1 jobservice[21101]: #011/usr/local/go/src/runtime/panic.go:500 +0x1a1

**Comment by ywk253100** [comment]
@timgriffiths  @wuzhihui1123 @zhoulouzi It's a known issue for replicating multiple namespace images. It has been fixed in the master code and will be available in the next release.

**Comment by zhaogj1987** [comment]
这个啥时会发新版本出来！！？目前我也遇到了相同的问题。

**Comment by ywk253100** [comment]
Fixed in 1.2.0.
